src/storage/txn/process.rs

```diff
@@ -77,34 +79,38 @@ impl Task {
     }
 }
 
-pub trait MsgScheduler: Clone + Send + 'static {
+pub trait MsgScheduler: Send + 'static {
     fn on_msg(&self, task: Msg);
+    fn clone(&self) -> Self;
```

Comment:
Why define our own `clone` method instead of using `Clone`?

Reply:
I couldn't derive `Clone` so implemented clone here, but manually implementing `Clone` is better.